### PR TITLE
Replace legacy image downloader with new actor-based implementation

### PR DESCRIPTION
**Pull request by azarovalex**
The current image downloader is based on `NSOperationQueue` and has some bugs that I previously described in https://github.com/mapbox/mapbox-navigation-ios/pull/4587. There is also a chance that these bugs caused memory corruption in some cases.

Instead of supporting sophisticated logic in legacy `NSOperation`-based implementation, I replaced it with new async/await implementation. It works the same way in fewer lines of code and because it's an actor, the compiler guarantees thread safety.

The only minor downside is that we can't use it on iOS 12, so I added another very simple implementation of the image downloader. It still caches responses but doesn't support reentrancy (meaning that 5 requests one after another will trigger 5 downloads), which is a trade-off we have to make.

TODO: 
- [x] add changelog entry

### CODE REVIEW

File: Sources/MapboxNavigation/ImageRepository.swift [hunk 1]
```diff
@@ -44,19 +43,14 @@ class ImageRepository {
             return
         }
 
-        let _ = imageDownloader.download(with: imageURL, completion: { [weak self] (cachedResponse, error) in
+        let _ = imageDownloader.download(with: imageURL, completion: { [weak self] result in
```
Comment (Udumft) [inline Sources/MapboxNavigation/ImageRepository.swift:46]:
```suggestion
        imageDownloader.download(with: imageURL, completion: { [weak self] result in
```
`download` returns nothing, so we can remove empty assignment

File: Sources/MapboxNavigation/ImageRepository.swift [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-            imageDownloader = ImageDownloader(sessionConfiguration: sessionConfiguration)
-        }
-    }
-
-    public static let shared = ImageRepository.init()
+    static let shared = ImageRepository()
 
     let imageCache: BimodalImageCache
-    fileprivate(set) var imageDownloader: ReentrantImageDownloader
-    private let requestTimeOut: TimeInterval = 10
```
Comment (Udumft) [inline Sources/MapboxNavigation/ImageRepository.swift:14]: Doesn't we lose this timeout setting?

**Review by Udumft — APPROVED** [review]
LGTM with few comments

File: Sources/MapboxNavigation/ImageRepository.swift (hunk 2 above)
Comment (azarovalex) [reply Sources/MapboxNavigation/ImageRepository.swift:14]: I don't see why we had it in the first place, imho we can remove it.

**Pull request merged**